xesmf/smm.py

```diff
@@ -6,19 +6,20 @@
 from pathlib import Path
 
 import numpy as np
-import scipy.sparse as sps
+import sparse as sps
 import xarray as xr
 
 
 def read_weights(weights, n_in, n_out):
     """
-    Read regridding weights into a scipy sparse COO matrix.
+    Read regridding weights into a DataArray (sparse COO matrix).
 
     Parameters
     ----------
-    weights : str, Path, xr.Dataset
+    weights : str, Path, xr.Dataset, xr.DataArray, scipy.sparse.coo_matrix, sparse.COO
```

Comment:
one last nitpick: `scipy.sparse.coo_matrix` do not seem supported by the code.
Do we want to add support with something like

```python
elif isinstance(weights, scipy.sparse.coo_matrix):
    # convert to dataarray
```

or remove from docstring?

Reply:
oh woups! That isinstance cal was the only place in all xesmf that required the explicit dependency to  `scipy`, so I preferred to remove it. I'll remove it from the docstring.